Evgeniy/subtask 3

St1ke opened the pull request.

St1ke committed:
[Subtask-3] Написал логику программы

WIP Не хватает метода обратной сортировки

St1ke committed:
[Subtask-3] Дополнил логику

1) Ожидаю three hundred review in my ass

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,50 @@
+        {
+            Console.WriteLine("Введите построчно слова для дальнейшей сортировки");
+            string userInput = "";
+            List<string> SetOfWords = new List<string>();
```
Зачем с большой буквы? Локальные переменные начинаются с маленькой.

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,50 @@
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите построчно слова для дальнейшей сортировки");
+            string userInput = "";
```
Зачем эта переменная снаружи цикла?

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,50 @@
+            Console.WriteLine("Введите построчно слова для дальнейшей сортировки");
+            string userInput = "";
+            List<string> SetOfWords = new List<string>();
+            bool validUserInput = false;
```
Не очень хорошее название - ввод то в любой случае валидный.

St1ke replied:
endOfCycle пойдёт сгодится?

Tauders replied:
Лучше тогда уж `isInputEnded` или `endOfInput`.

Tauders replied:
Лучше давать названия как можно ближе к теме происходящего.

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,50 @@
+                }
+            }
+            Console.WriteLine("Введите asc или desc для выбора сортировки");
+            string sortSelection = Console.ReadLine();
```
Переделай на использование `switch-case` здесь будет чуть лучше.

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,50 @@
+            if (sortSelection.ToLower() == "asc")
+            {
+                SetOfWords.Sort();
+                foreach (var word in SetOfWords)
+                {
+                    Console.WriteLine(word);
+                }
```
Этот кусок дублируется.

St1ke replied:
А что я могу с ним сделать, уже в switch/case, такой маленький кусок кода в метод чтоль?

Tauders replied:
Нет, можно иначе убрать это дублирование

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,50 @@
+            }
+            else
+            {
+                Console.WriteLine("Вы ввели неверное значение, значит вам нужно отдохнуть и больше не вводить значений");
```
А разреши ещё раз вводить значение.

Tauders reviewed (changes requested):
Задание сразу с звёздочкой.
Начни для `asc` и `desc` использовать Enum.

St1ke committed:
[Subtask-3] Поработал над вашими ревью:

1)Переименовал переменные
2)Добавил использование switch/case
WIP; Enum и повторение кода

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,56 @@
+                            Console.WriteLine(word);
+                        }
+                        break;
+                    case "exit":
+                        endOfInput = true;
+                        break;
```
Неверно по смыслу задачи - нигде не упоминается, что можно выйти. Если хочешь это добавить, то:
1) обсуди с нами (как с состовителями заданий)
2) поменяй название переменной
3) добавь указание пользователю на этот функционал

St1ke replied:
Ты просил разрешить ещё раз вводить значение,

Tauders replied:
И как это связано с добавлением функции выхода?

St1ke committed:
[Subtask-3] Убрал лишний функционал, добавил повторный ввод при неверном

St1ke committed:
[Subtask-3] Убрал дублирование кода

St1ke committed:
[Subtask-3] Ввёл Enum'ы

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,60 @@
+            }
+
+        }
+        enum SortWords
+        {
+            asc,
+            desc
+        }
```
Добавь значение по умолчанию

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,60 @@
+            }
+
+        }
+        enum SortWords
```
Название не очень

Tauders replied:
Перемести в отдельный файл

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,60 @@
+            bool isValidAction = false;
+            while (!isValidAction)
+            {
+                string sortSelection = Console.ReadLine().ToLower();
+                Enum.TryParse(sortSelection, out SortWords convertedSortSelection);
```
Убери `ToLower()` - посмотри, как можно использовать `TryParse` ещё.

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,60 @@
+        }
+        enum SortWords
+        {
+            asc,
```
Добавь числовые значения у них (типа `acs = 1` и т.д.), чтобы было однозначное определение.

St1ke committed:
[Subtask-3] Переименовал переменные, добавил неизвестное значение в Enum

St1ke committed:
[Subtask-3] Убрал метод ToLower(), добавил игнор регистра в TryParse

St1ke committed:
[Subtask-3] Перенёс enum в отдельный файл

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,59 @@
+                Enum.TryParse(sortSelection, true, out SortMethod convertedSortSelection);
+                switch (convertedSortSelection)
+                {
+                    case SortMethod.Unknown:
+                        Console.WriteLine("Вы выбрали неверный способ сортировки, попробуйте снова");
+                        break;
```
Не уверен, что надо обрабатывать по особому это.

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,59 @@
+            while (!isValidAction)
+            {
+                string sortSelection = Console.ReadLine();
+                Enum.TryParse(sortSelection, true, out SortMethod convertedSortSelection);
```
`convertedSortSelection` можно назвать `selectedSortMethod`

Tauders commented on Subtasks/Evgeniy/Subtask-3/Subtask-3/Program.cs:
```diff
@@ -0,0 +1,59 @@
+            {
+                Console.WriteLine(word);
+            }
+
```
Добавь `Все значения выведены`.

St1ke committed:
[Subtask-3]Переименовал переменную, убрал доп обработку
